examples/bri-3/src/bri/zeroKnowledgeProof/components/blockchain/blockchain.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import { ZeroKnowledgeProofVerificationInput } from '../../models/zeroKnowledgeProofVerificationInput';
+
+@Injectable()
+export class BlockchainService implements OnModuleInit {
```

Comment:
its only job is to talk to the blokchain so it should not know any of the business logic associated with the bpi.  this means basically having read and store methods, to read values from a smart contact and to trigger storage to the smart contact used for anchor hash storage onchain. Everything else above that layerr, such as verification of the hash existence, happens in the an agent which is relevant for the commands. Example:

VerifyAchorHashExistenceCommandHandler(private readonly agent ccsmAnchorHashStoragent) {
	agent.ThrowIfAchorHashDoesNotExist(anchorHash)
}

ThrowIfAchorHashDoesNotExist(string targetValue) {
  const anchorHash = this.blockchainService.read(targetValue);
  If(!anchorHash) {
  	throw NotFoundException(...)
  }
}